tools/ci/src/main.rs

```diff
@@ -0,0 +1,172 @@
+use anyhow::Result;
+use clap::{CommandFactory, Parser, Subcommand};
+use duct::cmd;
+use std::collections::HashMap;
+use std::env;
+
+#[derive(Parser)]
+#[command(
+    name = "spacetimedb-ci",
+    about = "SpacetimeDB CI tasks",
+    subcommand_required = false,
+    arg_required_else_help = false
+)]
+struct Cli {
+    #[command(subcommand)]
+    cmd: Option<CiCmd>,
+
+    #[arg(long)]
+    skip: Vec<String>,
+}
+
+#[derive(Subcommand)]
+enum CiCmd {
+    Test,
+    Lints,
+    WasmBindings,
+    Smoketests {
+        #[arg(trailing_var_arg = true)]
+        args: Vec<String>,
+    },
+    UpdateFlow {
+        #[arg(long)]
+        target: Option<String>,
+        #[arg(long, default_value = "true")]
+        github_token_auth: bool,
+    },
+    UnrealTests,
+    CliDocs,
+}
+
+macro_rules! run {
+    ($cmdline:expr) => {
+        run_command($cmdline, &Vec::new())
+    };
+    ($cmdline:expr, $envs:expr) => {
+        run_command($cmdline, $envs)
+    };
+}
+
+fn run_all_clap_subcommands(skips: &[String]) -> Result<()> {
+    let subcmds = Cli::command()
+        .get_subcommands()
+        .map(|sc| sc.get_name().to_string())
+        .collect::<Vec<_>>();
+
+    for subcmd in subcmds {
+        if skips.contains(&subcmd) {
+            log::info!("skipping {subcmd} as requested");
+            continue;
+        }
+        log::info!("executing cargo ci {subcmd}");
+        run!(&format!("cargo ci {subcmd}"))?;
+    }
+
+    Ok(())
+}
+
+fn run_command(cmdline: &str, additional_env: &[(&str, &str)]) -> Result<()> {
+    let mut env = env::vars().collect::<HashMap<_, _>>();
+    env.extend(additional_env.iter().map(|(k, v)| (k.to_string(), v.to_string())));
+    log::debug!("$ {cmdline}");
+    let status = cmd!("bash", "-lc", cmdline).full_env(env).run()?;
+    if !status.status.success() {
+        let e = anyhow::anyhow!("command failed: {cmdline}");
+        log::error!("{e}");
+        return Err(e);
+    }
+    Ok(())
+}
+
+fn main() -> Result<()> {
+    let cli = Cli::parse();
+
+    match cli.cmd {
+        Some(CiCmd::Test) => {
+            run!("sudo mkdir -p /stdb && sudo chmod 777 /stdb")?;
```

Comment:
I think this should be part of the CI still? I'm pretty sure we don't want to do this if running it on local machines.